### PR TITLE
Access report fixes

### PR DESCRIPTION
**Pull request by arodier**
It was too long since I did not check if this was working, and it is not easy to test, yet.

- Send email and xmpp alerts on connections denied
- Send alerts only once
- Allow to use whitelisting scheme per user

The last feature, allows one user account to access email connections only from certain addresses, like LAN. Pretty paranoid, I know.

### CODE REVIEW

File: install/playbooks/roles/access-check/templates/postmaster-ip-whitelist.cf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Postmaster specific whitelist file.
+# Lines starting with '#' are ignored
+# You can use ipcalc or whois tools to create CIDR notations
+
+# Whitelist LAN and local IP addresses
+192.168.0.0/16
+172..16.0.0/12
+10.10.0.0/8
```
Comment (fredericmoulins) [inline install/playbooks/roles/access-check/templates/postmaster-ip-whitelist.cf]: Typos: `172.16.0.0/12` and `10.0.0.0/8`.

File: install/playbooks/roles/dovecot/files/access-warn.sh [hunk 2]
```diff
@@ -91,44 +93,43 @@ MSG="${MSG}\n\nIP Details: https://duckduckgo.com/?q=whois+${IP}"
...
     if [ "$xmppOutput" != "" ]; then
         logger -p user.warning "sendxmpp error when sending warning from postmaster to $ALERT_ADDRESS: $xmppOutput"
     fi
+fi
 
-    # Send an email alert as well
-    subject="Alert from postmaster ($domain)"
-    from="postmaster@${domain}"
+# Send the alerts to an external / global account
+if [ "$ALERT_ADDRESS" != "" -a "$ALERT_ADDRESS" != "$EMAIL_ADDRESS" ]; then
```
Comment (fredericmoulins) [inline install/playbooks/roles/dovecot/files/access-warn.sh]: I think shellcheck prefers separate tests and `&&` or `||` logical operators.

File: install/playbooks/roles/dovecot/files/dovecot-postlogin [hunk 3]
```diff
@@ -174,6 +181,12 @@ for script in $login_scripts; do
         continue
     fi
 
+    # Skip this script, do not do anything sensitive if the connection is denied
+    if [ "$STATUS" = "DENIED" -a "$needDecriptKey" = "1" ]; then
```
Comment (fredericmoulins) [inline install/playbooks/roles/dovecot/files/dovecot-postlogin]: Maybe here also.

File: install/playbooks/roles/access-check/templates/postmaster-ip-whitelist.cf (hunk 1 above)
Comment (arodier) [reply install/playbooks/roles/access-check/templates/postmaster-ip-whitelist.cf]: Thanks

**Pull request merged**